### ISSUE TITLE
The code in semantic segmentation phase

### ISSUE THREAD

**Issue by ChunmengLiu1** [weixuansun/InferCam#2]
Hello, thanks for the great job!
I'm interesting in the part in segmentation network loss, which means the saliency map loss. Would you mind provide additional detailed information (the value of params) for me? Thanks a lot! 
Best wishes to you!

**Comment by weixuansun** [comment]
Hi, thank you for the question, the code for saliency subsidiary loss is here:

    output = self.model(image) 
    sal_pred = ((F.softmax(output, dim=1))[:,0,:,:])
    saliency_map_binary = 1-(saliency_map/255) 
    pseudo_bkg = (target==0)
    bkg_alignment = (pseudo_bkg == saliency_map_binary)
    sal_weight = (torch.sum(bkg_alignment, dim=(1,2))).double()/(target.shape[1]*target.shape[2])
    sal_loss = F.binary_cross_entropy(sal_pred.float(), saliency_map_binary.float(), reduction='none').mean(dim=(1,2))
    sal_loss = ((torch.exp(sal_weight)-1) * sal_loss).mean()
    seg_loss = self.criterion(output, target)
    loss = 0.1*sal_loss + seg_loss

I will update the repo and include this part of code soon.

**Comment by ChunmengLiu1** [comment]
&nbsp;Thanks!



---Original---
From: ***@***.***&gt;
Date: Wed, Mar 16, 2022 14:16 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [weixuansun/InferCam] The code in semantic segmentation phase (Issue #2)




 
Hi, thank you for the question, the code for saliency subsidiary loss is here:
 output = self.model(image)  sal_pred = ((F.softmax(output, dim=1))[:,0,:,:]) saliency_map_binary = 1-(saliency_map/255)  pseudo_bkg = (target==0) bkg_alignment = (pseudo_bkg == saliency_map_binary) sal_weight = (torch.sum(bkg_alignment, dim=(1,2))).double()/(target.shape[1]*target.shape[2]) sal_loss = F.binary_cross_entropy(sal_pred.float(), saliency_map_binary.float(), reduction='none').mean(dim=(1,2)) sal_loss = ((torch.exp(sal_weight)-1) * sal_loss).mean() seg_loss = self.criterion(output, target) loss = 0.1*sal_loss + seg_loss  
I will update the repo and include this part of code soon.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
